### PR TITLE
Create mdbook documentation

### PR DESCRIPTION
**Pull request by sharkAndshark**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -10,3 +10,4 @@ test_log*
 pg_data/
 config.yml
 tests/output/
+docs/book
```
Comment (nyurik) [inline .gitignore]: please enable "auto newline at the end of a file" in your IDE, e.g. https://stackoverflow.com/questions/16761227/how-to-make-intellij-idea-insert-a-new-line-at-every-end-of-file

**Review by nyurik — COMMENTED** [review]
Looks like a great start, thanks!  I added a justfile target for it to simplify things and pushed it to your branch. Looking forward to having a proper docs soon!

File: .gitignore (hunk 1 above)
Comment (sharkAndshark) [reply .gitignore]: Yeah. I didn't know this and enable it on vscode now. I'm trying to understand why it's important by google. Thx  :D

**Comment by sharkAndshark** [comment]
Do you think we need adjust the structure of book? @nyurik

File: .gitignore (hunk 1 above)
Comment (nyurik) [reply .gitignore]: why its important to have last new line? Same reason as that we want trailing commas in a list, even though last comma is optional:  if you make a change to the code, e.g. add one more line, you want the `diff` to only show one line change, not two

**Comment by nyurik** [comment]
> Do you think we need adjust the structure of book? @nyurik

Sure, any restructuring is possible - but I think we should start with matching the current readme, migrate to mdbook (e.g. make it auto-publish to the maplibre.org/martin site), and then start changing - otherwise we will have two duplicate places to keep up to date, and that's never good.

**Comment by sharkAndshark** [comment]
Yeah.One PR for one thing

**Review by nyurik — APPROVED** [review]
thank you !!  merging

**Pull request merged**